src/main/java/bitbucketpullrequestbuilder/bitbucketpullrequestbuilder/bitbucket/ApiClient.java

```diff
@@ -193,6 +184,23 @@ public void updatePullRequestComment(String pullRequestId, String content, Strin
         return null;
     }
 
+    private <T> List<T> getAllValues(String rootUrl, int pageLen, Class<T> cls) {
+        List<T> values = new ArrayList<T>();
+        try {
+            String url = rootUrl + "?pagelen=" + pageLen;
+            do {
+                final JavaType type = TypeFactory.defaultInstance().constructParametricType(Pullrequest.Response.class, cls);
+                Pullrequest.Response<T> response = parse(get(url), type);
+                values.addAll(response.getValues());
+                url = response.getNext();
+            } while (url != null);
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "invalid response.", e);
```

Comment:
Nit: you could add the parameterized class name to the error string to make debugging a little easier.